packages/rx_bloc_test/README.md

```diff
@@ -40,4 +42,54 @@ rxBlocTest<DetailsBloc,String>(
   skip: 3, // This will skip first two values + the initial value
   expect: ['Hello world'],
 )
-```
+```  
+
+## rxBlocFakeAsyncTest
+For the most complex test cases with multiple events, which contain `throttleTime` or/and `debounceTime` or similar inside, and to speed up test execution,`rxBlocFakeAsyncTest` should be used. It contains an instance of [FakeAsync](https://pub.dev/packages/fake_async) inside `act`, which provides a way to fire all asynchronous events that are scheduled for that time period without actually needing the test to wait for real time to elapse.
+
+**BAD:**
+```dart
+rxBlocTest<TimelineBloc, PaginatedList<SnapshotModel>>(
+      'TimelineBloc - CoordinatorBloc Update Timeline',
+      build: () async {
+        return _createBlocInstance();
+      },
+      state: (bloc) => bloc.states.paginatedList,
+      act: (bloc) async {
+        await Future.delayed(const Duration(milliseconds: 500));
+        coordinatorBloc.events.updateTimeline();
+        await Future.delayed(const Duration(milliseconds: 500));
+      },
+      expect: [
+        Snapshot.emptyList,
+        Snapshot.emptyListLoading,
+        Snapshot.notEmptyList,
+      ],
+    );
+ ```
+
+**GOOD:**
+```dart
+rxBlocFakeAsyncTest<GlobalSearchBlocType, Result<List<GlobalSearchResultGroupModel>?>>(
+    'GlobalSearchBloc - task updated',
+    state: (bloc) => bloc.states.searchResults,
+    build: () {
+      _setUpMocks(response: GlobalSearchStub.taskSearchResponse);
```

Comment:
This is very appointed approach, which not anybody will understand. Can you please return a CounterBloc  instance and just add a comment that here all BloC's dependencies can be mocked using mockito.